security/ipsw/files/reproducible-build.diff

```diff
@@ -1,47 +1,47 @@
 diff -urN a/cmd/ipsw/cmd/class_dump.go b/cmd/ipsw/cmd/class_dump.go
---- a/cmd/ipsw/cmd/class_dump.go	2024-03-05 21:03:07.000000000 +0000
-+++ b/cmd/ipsw/cmd/class_dump.go	2024-03-18 23:19:42.000000000 +0000
+--- a/cmd/ipsw/cmd/class_dump.go	2024-03-30 19:55:36.000000000 +0000
++++ b/cmd/ipsw/cmd/class_dump.go	2024-03-31 21:45:28.000000000 +0000
 @@ -132,7 +132,7 @@
  			Headers:     viper.GetBool("class-dump.headers"),
  			ObjcRefs:    viper.GetBool("class-dump.refs"),
  			Deps:        viper.GetBool("class-dump.deps"),
--			IpswVersion: fmt.Sprintf("Version: %s, BuildTime: %s", strings.TrimSpace(AppVersion), strings.TrimSpace(AppBuildTime)),
+-			IpswVersion: fmt.Sprintf("Version: %s, BuildCommit: %s", strings.TrimSpace(AppVersion), strings.TrimSpace(AppBuildCommit)),
```

Comment:
What happens if you don't patch out this BuildCommit stuff?

Reply:
I ask because after seeing that you were patching out the BuildTime stuff I filed the upstream bug report to get them to remove it. Their response was to change it to BuildCommit.

Your original reproducible-build.diff patch made the build reproducible by removing build time information. But now the build should already be reproducible without needing to patch it.

Reply:
[I've expressed my dislike of something like this when I first submitted this port.](https://github.com/macports/macports-ports/pull/23071#discussion_r1530651378)

To give an example: with previous versions, it made sense to patch out `BuildTime` as having it be a constant (like "unknown") would provide no insightful information in the version message. So rather than having it look like

```
$ ipsw version
Version: 3.1.456, BuildTime: unknown
```

the patch made it look like

```
$ ipsw version
Version 3.1.456
```

`BuildCommit` suffers from this as well. While I was made aware of the issue filed, it's not in the best interest for MacPorts to dictate how software is developed upstream, unless it's a bug with the software itself.

A simple flag that sets `BuildCommit` to "unknonwn" replaces the need of the patch; I'll move forward with that approach, but it was never necessary to have it be changed from `BuildTime` to `BuildCommit`.